stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/ImageAttachmentsGroupView.kt

```diff
@@ -101,21 +107,36 @@ internal class ImageAttachmentsGroupView : ConstraintLayout {
             constrainViewToParentBySide(imageAttachmentView, ConstraintSet.TOP)
             constrainViewToParentBySide(imageAttachmentView, ConstraintSet.BOTTOM)
 
-            val imageWidth = first.originalWidth?.toFloat()
-            val imageHeight = first.originalHeight?.toFloat()
-
             // Used to set a dimension ratio before we load an image
             // so that message positions don't jump after we load it.
-            if (imageWidth != null && imageHeight != null) {
-                val ratio = (imageWidth / imageHeight).toString()
-                this.setDimensionRatio(imageAttachmentView.id, ratio)
+            if (imageAspectRatio != 0f) {
+                this.setDimensionRatio(imageAttachmentView.id, imageAspectRatio.toString())
             } else {
                 constrainHeight(imageAttachmentView.id, LayoutParams.WRAP_CONTENT)
             }
 
             applyTo(this@ImageAttachmentsGroupView)
         }
 
+        if (imageAspectRatio != 0f) {
+            imageAttachmentView.post {
+                imageAttachmentView.binding.imageView.scaleType =
+                    if (imageAttachmentView.measuredHeight < maxImageAttachmentHeight) {
+                        ImageView.ScaleType.FIT_XY
+                    } else {
+                        val scaleFactor = imageAttachmentView.measuredWidth.toFloat() / imageWidth!!
```

Comment:
Also, I think it would be good if we could get rid of the double bangs here.

I understand the reasoning behind them but it introduces possible unforeseen edgecases, such as the one we've seen now where `NaN` would evaluate to different than `0f` and could possibly lead to an `NPE` in the case of large images hosted on CDNs which do not report image dimensions.

Reply:
refactored to a funcion which will take care of checking for nul and eliminate !!.